architecture/ambient/lifecycle.md

```diff
@@ -0,0 +1,136 @@
+# Ztunnel Lifecyle
+
+This document describes the lifecycle of various aspects of Ztunnel, and its relationship to workloads and the CNI.
+
+At a high level, our goal is to provide complete connectivity to a workload throughout its entire lifetime.
+Failing to do so can be an availability risk (if we deny traffic that should succeed) or a security risk (if we allow traffic that should be denied).
+
+## High level overview
+
+At a high level, the relevant components look as such:
+
+```mermaid
+flowchart TD
+    CNIP["CNI Plugin"]
+    CNI["CNI Agent"]
+    Z["Ztunnel"]
+    Pods["Pods"]
+    CNI --Programs--> Z
+    CNI --Watches--> Pods
+    Z --Runs Within--> Pods
+    CNIP --Calls--> CNI
+```
+
+The CNI Plugin is a binary installed as a [CNI plugin](https://kubernetes.io/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins/) on the node.
+Kubernetes is responsible for invoking this when a Pod is being started (before the containers run).
+When this occurs, the plugin will call out to the CNI Agent to program the network.
```

Comment:
K8S will call the Istio CNI binary (path = ??? ) , which will use a UDS socket ( path = ???) to communicate the pod info to the server running in the Istio CNI Daemon. 

The Istio CNI daemon will keep an in-memory cache with all the pods and their corresponding network namespaces.